### PR TITLE
Authentication bugs

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)
Areas to be tested for appropriate response:
- [x] Android
  - [x] With internet connection:
    - [x] Email Sign up
    - [x]   Phone number Sign up
    - [x]   Account deletion
    - [x]   Email login
    - [x]   Phone number login
   - [x] Without internet connection:
      - [x] Email Sign up
      - [x]   Phone number Sign up
      - [x]   Account deletion
      - [x]   Email login
      - [x]   Phone number login
- [x] iOS
  - [x] With internet connection:
    - [x] Email Sign up
    - [x]   Phone number Sign up
    - [ ]   Account deletion
    - [ ]   Email login
    - [ ]   Phone login
   - [x] Without internet connection:
      - [ ] Email Sign up
      - [ ]   Phone number Sign up
      - [ ]   Account deletion
      - [ ]   Email login
      - [ ]   Phone number login

  
#### Status of maturity (all need to be checked before merging):

- [ ] I've tested this locally
- [ ] I consider this code done
- [ ] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

### CODE REVIEW

**Review by BenjaminSsempala — APPROVED** [review]

File: mobile/lib/blocs/phone_verification/phone_verification_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import 'package:app/models/models.dart';
+import 'package:equatable/equatable.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+part 'phone_verification_event.dart';
+part 'phone_verification_state.dart';
+
+class PhoneVerificationBloc
+    extends Bloc<PhoneVerificationEvent, PhoneVerificationState> {
```
Comment (NoahNsimbe) [inline mobile/lib/blocs/phone_verification/phone_verification_bloc.dart:9]: verification blocs

File: mobile/lib/models/enum_constants.dart [hunk 2]
```diff
@@ -8,6 +8,12 @@ part 'enum_constants.g.dart';
 
 enum Environment { dev, prod }
 
+enum AuthenticationStatus {
+  initial,
+  error,
+  success;
+}
```
Comment (NoahNsimbe) [inline mobile/lib/models/enum_constants.dart:15]: authentication states

File: mobile/lib/screens/settings/account_deletion_widgets.dart [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+class AccountDeletionTitle extends StatelessWidget {
+  const AccountDeletionTitle({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.only(left: 16, right: 16, top: 10),
+      child: AutoSizeText(
+        "Enter code to delete account",
+        textAlign: TextAlign.center,
```
Comment (NoahNsimbe) [inline mobile/lib/screens/settings/account_deletion_widgets.dart:17]: requesting for a verification code before deleting account

File: mobile/lib/themes/app_theme.dart [hunk 4]
```diff
@@ -497,6 +498,54 @@ ThemeData customTheme() {
         aqiMaroonTextColor: Color(0xffDBA5B2),
       ),
     ],
+    inputDecorationTheme: ThemeData.light().inputDecorationTheme.copyWith(
+          contentPadding: const EdgeInsets.fromLTRB(16, 12, 0, 12),
```
Comment (NoahNsimbe) [inline mobile/lib/themes/app_theme.dart:502]: global text themes

**Pull request merged**